src/main/java/uk/ac/ebi/eva/commons/mongodb/entity/subdocuments/VariantSourceEntryMongo.java

```diff
@@ -0,0 +1,272 @@
+
+    public final static String FORMAT_FIELD = "fm";
+
+    public final static String SAMPLES_FIELD = "samp";
+
+    public static final String DEFAULT = "def";
+
+    @Field(FILEID_FIELD)
+    private String fileId;
+
+    @Field(STUDYID_FIELD)
+    private String studyId;
+
+    @Field(ALTERNATES_FIELD)
+    private String[] alternates;
+
+    @Field(ATTRIBUTES_FIELD)
+    private BasicDBObject attributes;
+
+    @Field(FORMAT_FIELD)
+    private String format;
+
+    @Field(SAMPLES_FIELD)
+    private Map<String, Object> samples;
+
+    VariantSourceEntryMongo() {
+        // Spring empty constructor
+    }
+
+    public VariantSourceEntryMongo(IVariantSourceEntry variantSourceEntry) {
+        this(
+                variantSourceEntry.getFileId(),
+                variantSourceEntry.getStudyId(),
+                variantSourceEntry.getSecondaryAlternates(),
+                variantSourceEntry.getAttributes(),
+                variantSourceEntry.getFormat(),
+                variantSourceEntry.getSamplesData()
+        );
+    }
+
+    public VariantSourceEntryMongo(String fileId, String studyId, String[] alternates, Map<String, String> attributes) {
+        this(fileId, studyId, alternates, attributes, null, null);
+    }
+
+    public VariantSourceEntryMongo(String fileId, String studyId, String[] alternates, Map<String, String>
+            attributes, String format, List<Map<String, String>> samplesData) {
+        this.fileId = fileId;
+        this.studyId = studyId;
+        if (alternates != null && alternates.length > 0) {
+            this.alternates = new String[alternates.length];
+            System.arraycopy(alternates, 0, this.alternates, 0, alternates.length);
+        }
+
+        if (attributes != null) {
+            this.attributes = buildAttributes(attributes);
+        }
+
+        if (format == null || samplesData == null || samplesData.isEmpty()) {
+            this.format = null;
+            this.samples = null;
+        } else {
+            this.format = format;
+            this.samples = buildSampleData(samplesData);
+        }
+    }
+
+    private BasicDBObject buildSampleData(List<Map<String, String>> samplesData) {
+        Map<Genotype, List<Integer>> genotypeCodes = classifySamplesByGenotype(samplesData);
+
+        // Get the most common genotype
+        Map.Entry<Genotype, List<Integer>> longestList = getLongestGenotypeList(genotypeCodes);
+
+        // In Mongo, samples are stored in a map, classified by their genotype.
+        // The most common genotype will be marked as "default" and the specific
+        // positions where it is shown will not be stored. Example from 1000G:
+        // "def" : 0|0,
+        // "0|1" : [ 41, 311, 342, 358, 881, 898, 903 ],
+        // "1|0" : [ 262, 290, 300, 331, 343, 369, 374, 391, 879, 918, 930 ]
+        BasicDBObject mongoSamples = new BasicDBObject();
+        for (Map.Entry<Genotype, List<Integer>> entry : genotypeCodes.entrySet()) {
```

Comment:
The webservice can be quite slow for large query ranges, I would argue that it would be best to increase performance wherever possible. For one variant this might not take much longer but when the resultset has thousands of results it might mount up. I also don't think that this:

```
    private BasicDBObject buildSampleData(List<Map<String, String>> samplesData) {
        Map<Genotype, List<Integer>> genotypeCodes = classifySamplesByGenotype(samplesData);

        String modeGtStr = null;
        int modeGtCount = 0;

        BasicDBObject mongoSamples = new BasicDBObject();
        for (Map.Entry<Genotype, List<Integer>> entry : genotypeCodes.entrySet()) {
            String genotypeStr = entry.getKey().generateDatabaseString();
            List<Integer> sampleIndexes = entry.getValue();

            mongoSamples.append(genotypeStr, sampleIndexes);
            if (sampleIndexes.size() > modeGtCount) {
                modeGtStr = genotypeStr;
                mongoSamples.append(DEFAULT, genotypeStr);
            }
        }
        mongoSamples.remove(modeGtStr);
        return mongoSamples;
    }
```

is that much less readable.